shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.mm

```diff
@@ -70,6 +72,75 @@ - (BOOL)pointInside:(CGPoint)point withEvent:(UIEvent*)event {
   return NO;
 }
 
+// Creates and initializes a UIVisualEffectView with a UIBlurEffect. Extracts and returns its
+// gaussianFilter. Logs errors and returns if Apple's API has changed and the filter can't be
+// extracted.
+- (NSObject*)extractGaussianFilter {
+  UIVisualEffectView* visualEffectView = [[UIVisualEffectView alloc]
+      initWithEffect:[UIBlurEffect effectWithStyle:UIBlurEffectStyleLight]];
+
+  UIView* view = [visualEffectView.subviews firstObject];
+  if (!view || ![view isKindOfClass:NSClassFromString(@"_UIVisualEffectBackdropView")]) {
+    FML_DLOG(ERROR) << "Apple's API for UIVisualEffectView changed. Update the implementation to "
+                       "access its subviews.";
+    return nil;
+  }
+
+  NSObject* gaussianFilter = [[view.layer.filters firstObject] retain];
+  if (!gaussianFilter || ![[gaussianFilter valueForKey:@"name"] isEqual:@"gaussianBlur"]) {
+    FML_DLOG(ERROR) << "Apple's API for UIVisualEffectView changed. Update the implementation to "
+                       "access the Gaussian blur filter. ";
+    return nil;
+  }
+
+  [visualEffectView release];
+
+  if (![[gaussianFilter valueForKey:@"inputRadius"]
+          isKindOfClass:[NSNumber class]]) {  // TODO EMILY: is there another way to check that
+                                              // inputRadius key is valid? -> DOCUMENT ATTEMPTS
+    FML_DLOG(ERROR) << "Apple's API for UIVisualEffectView changed. Update the implementation "
+                       "access the Gaussian blur filter's properties.";
+    return nil;
+  }
+
+  return gaussianFilter;
+}
+
+- (void)applyBackdropFilters:(NSArray*)blurRadii {
+  NSObject* gaussianFilter;
+  if (!_gaussianFilters) {
+    _gaussianFilters = [[[NSMutableArray alloc] init] retain];
+  }
+
+  if ([blurRadii count] > [_gaussianFilters count]) {
+    gaussianFilter = [self extractGaussianFilter];
+    if (!gaussianFilter)
+      return;
+  }
+
+  if ([blurRadii count] < [_gaussianFilters count]) {
+    [_gaussianFilters removeObjectsInRange:(NSRange){[blurRadii count],
+                                                     [_gaussianFilters count] - [blurRadii count]}];
+  }
+
+  for (int i = 0; i < (int)[blurRadii count]; i++) {
```

Comment:
[blurRadii count] doesnt need to be casted explicitly. you can use NSInteger in loops:
for (NSInteger i = 0; i < [blurRadii count]; i++) {

Reply:
We are adding new filters and setting filter values in the same for loop.
How about separate them into 2 for loops?
The first loop adds necessary additional filters,
the second loop sets the values.